extension/src/cli/git/reader.ts

```diff
@@ -105,6 +101,22 @@ export class GitReader extends GitCli {
     }
   }
 
+  public async getCurrentBranch(cwd: string): Promise<string> {
+    const options = getOptions(cwd, Command.BRANCH)
+    try {
+      const branches = await this.executeProcess(options)
+      return (
+        trimAndSplit(branches)
```

Comment:
Yes, that was the only way to make it work for that. Since detached HEAD is so much present in our demo project, I figured that's a common state that needs to be covered.